src/javascript/SelectorTypes/DateTimePicker/DateTimePicker.jsx

```diff
@@ -16,13 +16,49 @@ export const DateTimePicker = ({id, field, value, editorContext, onChange}) => {
     const isDateTime = variant === 'datetime';
     const disabledDays = fillDisabledDaysFromJCRConstraints(field, isDateTime);
 
-    // Handle the date format, only "en" have a specific format.
-    let displayDateFormat = editorContext.uilang === 'en' ? 'MM/DD/YYYY' : 'DD/MM/YYYY';
-    displayDateFormat = isDateTime ? (displayDateFormat + ' HH:mm') : displayDateFormat;
-    const displayDateMask = isDateTime ? '__/__/____ __:__' : '__/__/____';
+    const userNavigatorLocale = editorContext.navigatorLocale;
+
+    // Handle the date format, some languages have a specific format that we can manage here.
+    // Please note that a single digit format is not supported by our DatePicker, e.g. 'D/M/YYYY' or 'D.M.YYYY'.
+    const specificDateFormat = {
+        'de-AT': 'DD.MM.YYYY',
+        'de-CH': 'DD.MM.YYYY',
+        'de-DE': 'DD.MM.YYYY',
+        'de-LI': 'DD.MM.YYYY',
+        'de-LU': 'DD.MM.YYYY',
+        'en-PH': 'MM/DD/YYYY',
+        'en-US': 'MM/DD/YYYY',
+        'en-ZW': 'MM/DD/YYYY',
+        'es-PA': 'MM/DD/YYYY',
+        'es-US': 'MM/DD/YYYY',
+        'zh-CN': 'YYYY/MM/DD'
+    };
```

Comment:
I would put that in a separate file to ease the update the list in case it's required in future.

Reply:
Done.